### PR TITLE
[Compute] Update from CoreOS to Flatcar

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Issue: #18484

**Description**

As CoreOS is deprecated and there is no Image in the marketplace any more, update to use Flatcar linux instead of it.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 1]
```diff
@@ -787,15 +787,16 @@ def _validate_vm_vmss_accelerated_networking(cli_ctx, namespace):
...
+        # Ubuntu 16.04 | 18.04, SLES 12 SP3, RHEL 7.4, CentOS 7.4, Flatcar Linux, Debian "Stretch" with backports kernel
         # Oracle Linux 7.4, Windows Server 2016, Windows Server 2012R2
         publisher, offer, sku = namespace.os_publisher, namespace.os_offer, namespace.os_sku
         if not publisher:
             return
         publisher, offer, sku = publisher.lower(), offer.lower(), sku.lower()
         distros = [('canonical', 'UbuntuServer', '^16.04|^18.04'),
                    ('suse', 'sles', '^12-sp3'), ('redhat', 'rhel', '^7.4'),
-                   ('openlogic', 'centos', '^7.4'), ('coreos', 'coreos', None), ('credativ', 'debian', '-backports'),
+                   ('openlogic', 'centos', '^7.4'), ('kinvolk', 'flatcar-container-linux-free', None),
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:798]: recommend to add actionable error message for value coreOS

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:804]: OK

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 2]
```diff
@@ -787,15 +787,16 @@ def _validate_vm_vmss_accelerated_networking(cli_ctx, namespace):
                 return
 
         # VMs need to be a supported image in the marketplace
-        # Ubuntu 16.04 | 18.04, SLES 12 SP3, RHEL 7.4, CentOS 7.4, CoreOS Linux, Debian "Stretch" with backports kernel
+        # Ubuntu 16.04 | 18.04, SLES 12 SP3, RHEL 7.4, CentOS 7.4, Flatcar Linux, Debian "Stretch" with backports kernel
```
Comment (ahrkrak) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:790]: We should use "Flatcar Container Linux" (or just "Flatcar" for short), but not "Flatcar Linux".

**Review by ahrkrak — APPROVED** [review]
One small suggested naming change. Otherwise LGTM.

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py (hunk 1 above)
Comment (ahrkrak) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:804]: suggest " please use Flatcar Container Linux instead." (add "Container" and delete "of it").

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py:804]: OK, done

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_validators.py]: OK, done

File: src/azure-cli/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -787,15 +787,22 @@ def _validate_vm_vmss_accelerated_networking(cli_ctx, namespace):
...
         publisher, offer, sku = namespace.os_publisher, namespace.os_offer, namespace.os_sku
         if not publisher:
             return
         publisher, offer, sku = publisher.lower(), offer.lower(), sku.lower()
+
+        if publisher == 'coreos' or offer == 'coreos':
+            from azure.cli.core.parser import InvalidArgumentValueError
+            raise InvalidArgumentValueError("As CoreOS is deprecated and there is no image in the marketplace any more,"
+                                            " please use Flatcar Container Linux instead.")
+
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_validators.py:801]: The actionable error message for CoreOS

**Review by Juliehzl — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: doc/install_linux_prerequisites.md [hunk 4]
```diff
@@ -93,6 +93,6 @@ Python 2.7.8 should be already on the machine. Install Python 3.6+ if needed.
 sudo zypper refresh && sudo zypper --non-interactive install gcc libffi-devel python-devel openssl-devel
 ```
 
-### CoreOS Stable-899.15.0 / Beta-1010.1.0 / Alpha-1010.1.0
+### Flatcar
 
-Python is not installed by default.
+Python is installed in the Azure-specific distribution of Flatcar, but is installed into the non-standard location /usr/share/oem/python/bin/python.
```
Comment (jiasli) [inline doc/install_linux_prerequisites.md:98]:
```suggestion
Python is installed in the Azure-specific distribution of Flatcar, but is installed into the non-standard location `/usr/share/oem/python/bin/python`.
```

Comment (jiasli) [reply doc/install_linux_prerequisites.md]: Given python is already installed on "Azure-specific distribution of Flatcar", shouldn't we provide the command to install Azure CLI dependencies?

File: doc/install_linux_prerequisites.md [hunk 5]
```diff
@@ -93,6 +93,6 @@ Python 2.7.8 should be already on the machine. Install Python 3.6+ if needed.
 sudo zypper refresh && sudo zypper --non-interactive install gcc libffi-devel python-devel openssl-devel
 ```
 
-### CoreOS Stable-899.15.0 / Beta-1010.1.0 / Alpha-1010.1.0
+### Flatcar
 
-Python is not installed by default.
+Python is installed in the Azure-specific distribution of Flatcar, but is installed into the non-standard location `/usr/share/oem/python/bin/python`.
```
Comment (jiasli) [inline doc/install_linux_prerequisites.md:98]: Given python is already installed on "Azure-specific distribution of Flatcar", shouldn't we provide the command to install Azure CLI dependencies?

Comment (zhoxing-ms) [reply doc/install_linux_prerequisites.md:98]: This description is provided by ahrkrak. @ahrkrak Hi, could you please help to have a look~

Comment (jiasli) [reply doc/install_linux_prerequisites.md:98]: Now I doubt if there is really someone to use Flatcar to install Azure CLI. 🤣

**Pull request merged**